spec/observables/generate-spec.ts

```diff
@@ -0,0 +1,77 @@
+import * as Rx from '../../dist/cjs/Rx.KitchenSink';
+import '../../dist/cjs/add/observable/generate';
+import {TestScheduler} from '../../dist/cjs/testing/TestScheduler';
+declare const {expectObservable};
+declare const rxTestScheduler: TestScheduler;
+
+const Observable = Rx.Observable;
+
+describe('Observable.generate', () => {
+  it('should complete if condition does not meet', () => {
+    const source = Observable.generate(1, x => false, x => x + 1);
+    const expected = '|';
+
+    expectObservable(source).toBe(expected);
+  });
+  it('should produce first value immediately', () => {
+    const source = Observable.generate(1, x => x == 1, x => x + 1);
+    const expected = '(1|)';
+
+    expectObservable(source).toBe(expected, { '1': 1 });
+  });
+  it('should produce all values synchronously', () => {
+    const source = Observable.generate(1, x => x < 3, x => x + 1);
+    const expected = '(12|)';
+
+    expectObservable(source).toBe(expected, { '1': 1, '2': 2 });
+  });
+  it('should use result selector', () => {
+    const source = Observable.generate(1, x => x < 3, x => x + 1, x => (x + 1).toString());
+    const expected = '(23|)';
+
+    expectObservable(source).toBe(expected);
+  });
+  /* the following test fails due to bug #1543 */
+  /*it('should allow omit condition', () => {
+    const source = Observable.generate({
+      initialState: 1,
+      iterate: x => x + 1,
+      resultSelector: x => x.toString()
+    }).take(5);
+    const expected = '(12345|)';
+
+    expectObservable(source).toBe(expected);
+  });*/
```

Comment:
I've checked in https://github.com/ReactiveX/rxjs/pull/1544 prior to this and believe this is now unblocked - would you mind to update this one?